### PR TITLE
fix Issue 15212 - BigInt should implement opCast!long and opCast!int

### PR DESCRIPTION
**Pull request by CyberShadow**
https://issues.dlang.org/show_bug.cgi?id=15212

### CODE REVIEW

File: std/bigint.d [hunk 1]
```diff
@@ -566,6 +566,71 @@ public:
...
+
+        import std.conv : ConvOverflowException;
+        import std.string : format;
+        throw new ConvOverflowException(
+            "BigInt(%d) cannot be represented as a %s"
+            .format(this, T.stringof));
+    }
+
+    ///
+    unittest
```
Comment (Hackerpilot) [inline std/bigint.d]: This unit test will become the example for `to!ulong`, but it doesn't have a call to to!ulong, just calls for int and ubyte.

**Comment by Hackerpilot** [comment]
:white_check_mark:

File: std/bigint.d (hunk 1 above)
Comment (CyberShadow) [inline std/bigint.d]: Yeah, the "why the heck not ALL integer types?" thought came after I filed the issue :)

**Comment by Hackerpilot** [comment]
I don't see anybody objecting to this, so I'm going to press the button.

**Comment by Hackerpilot** [comment]
Auto-merge toggled on

**Pull request merged**